tests/lib/test_wheel.py

```diff
@@ -80,7 +80,7 @@ def test_make_metadata_file_custom_value_overrides() -> None:
 
 def test_make_metadata_file_custom_contents() -> None:
     value = b"hello"
-    f = default_make_metadata(value=value)
+    f = default_make_metadata(value=value)  # type: ignore[arg-type]
```

Comment:
Do you know the reason of this one? Is that some regression in mypy?

Reply:
Prior to mypy 1.11, mypy did not understand functools.partial at all, so calls here were effectively going unchecked. mypy now struggles a little to solve the value constrained type var once it's been partially solved in the functools.partial call.

Looking at it a little closer, the type variable in `make_wheel_metadata_file` is actually unnecessary, so I could remove it to make one of these go away